src/ddsrt/include/dds/ddsrt/avl.h

```diff
@@ -200,7 +200,7 @@ typedef void (*ddsrt_avl_augment_t) (void *vnode, const void *left, const void *
 /**
  * @brief User defined walk function to perform an action on each node in the tree.
  */
-typedef void (*ddsrt_avl_walk_t) (void *vnode, void *arg);
+typedef void (*ddsrt_avl_walk_t) (const void *vnode, void *arg);
```

Comment:
This one is incorrect: the point of `ddsrt_avl_cwalk_t` defined immediately below is that there are two versions, one which is `const` and one which is not (and thereby allows the callback function to modify the contents within reason).